src/test/scala/freeeventsourcing/syntax/ProcessSyntaxTests.scala

```diff
@@ -0,0 +1,303 @@
+  }
+
+  "ProcessSyntax.firstOf " should " accept a timeout that returns a value and a selector" in {
+    val r = firstOf(_.
+      on(selectorClosed).event.
+      timeout(Instant.now)(Monad[ProcessMonad].pure("timeout")))
+    "r : ProcessMonad[Closed :+: String :+: CNil]" should compile
+  }
+
+  "ProcessSyntax.firstOf(_.selector.event) " should " should use one selectors and return the event" in {
+    val evt = Opened("Mario")
+    firstOf(_.
+      on(selectorOpened).event) should runFromWithResult(
+      Expect.firstOf(FirstOfSelector(selectorOpened))(0, evt)
+    )(Coproduct[Opened :+: CNil](evt))
+  }
+
+  "ProcessSyntax.firstOf(_.waitUntil.noop) " should " should use one selectors and return unit" in {
+    val instant = Instant.now()
+    firstOf(_.
+      timeout(instant)(noop)) should runFromWithResult(
+      Expect.firstOf(FirstOfUntil(instant))(0, ())
+    )(Coproduct[Unit :+: CNil](()))
+  }
+
+  "ProcessSyntax.firstOf(two _.selector.event).1 " should " should use two selectors and return the first event" in {
+    val evt = Closed()
+    firstOf(_.
+      on(selectorOpened).event.
+      on(selectorClosed).event) should runFromWithResult(
+      Expect.firstOf(
+        FirstOfSelector(selectorOpened),
+        FirstOfSelector(selectorClosed)
+      )(1, evt)
+    )(Coproduct[Opened :+: Closed :+: CNil](evt))
+  }
+
+  "ProcessSyntax.firstOf(two _.selector.event).2 " should " should use two selectors and return the second event" in {
+    val evt = Opened("Mario")
+    firstOf(_.
+      on(selectorOpened).event.
+      on(selectorClosed).event) should runFromWithResult(
+      Expect.firstOf(
+        FirstOfSelector(selectorOpened),
+        FirstOfSelector(selectorClosed)
+      )(0, evt)
+    )(Coproduct[Opened :+: Closed :+: CNil](evt))
+  }
+
+  "ProcessSyntax.firstOf(two _.selector.event and timeout).1 " should " should use two selectors, a timeout and return the first event" in {
+    val evt = Opened("Mario")
+    val instant = Instant.now()
+    firstOf(_.
+      on(selectorOpened).event.
+      on(selectorClosed).event.
+      timeout(instant)(noop)) should runFromWithResult(
+      Expect.firstOf(
+        FirstOfSelector(selectorOpened),
+        FirstOfSelector(selectorClosed),
+        FirstOfUntil(instant)
+      )(0, evt)
+    )(Coproduct[Opened :+: Closed :+: Unit :+: CNil](evt))
+  }
+
+  "ProcessSyntax.firstOf(two _.selector.event and timeout).2 " should " should use two selectors, a timeout and return the second event" in {
+    val evt = Closed()
+    val instant = Instant.now()
+    firstOf(_.
+      on(selectorOpened).event.
+      on(selectorClosed).event.
+      timeout(instant)(noop)) should runFromWithResult(
+      Expect.firstOf(
+        FirstOfSelector(selectorOpened),
+        FirstOfSelector(selectorClosed),
+        FirstOfUntil(instant)
+      )(1, evt)
+    )(Coproduct[Opened :+: Closed :+: Unit :+: CNil](evt))
+  }
+
+  //TODO more tests
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Take the required action to fix the issue indicated by this comment.](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2603375241/issues/source?bid=3298199&fileBranchId=3298202#l284)